From c929f2c1bdd173686b295a003e4f6dda5f0fa98a Mon Sep 17 00:00:00 2001
From: Robert Scheck <robert-scheck@users.noreply.github.com>
Date: Tue, 11 Feb 2025 01:48:07 +0100
Subject: [PATCH] Fix C23 build failures using GCC 15
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc -DHAVE_CONFIG_H -I.     -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection  -c -o signal.o signal.c
make[1]: Leaving directory '/builddir/build/BUILD/moon-buggy-1.0.51-build/moon-buggy-1.0.51'
signal.c: In function ‘install_signal’:
signal.c:56:21: error: assignment to ‘__sighandler_t’ {aka ‘void (*)(int)’} from incompatible pointer type ‘void (*)(void)’ [-Wincompatible-pointer-types]
   56 |   action.sa_handler = handler;
      |                     ^
In file included from signal.c:12:
/usr/include/signal.h:72:16: note: ‘__sighandler_t’ declared here
   72 | typedef void (*__sighandler_t) (int);
      |                ^~~~~~~~~~~~~~
---
 signal.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/signal.c b/signal.c
index ac16424..120b9d7 100644
--- a/signal.c
+++ b/signal.c
@@ -45,7 +45,7 @@ unblock (void)
 }
 
 static void
-install_signal (int signum, RETSIGTYPE (*handler) ())
+install_signal (int signum, RETSIGTYPE (*handler) (int))
 /* Emulate the `signal' function via `sigaction'.  */
 {
   struct sigaction  action;
